src/Evolution/Systems/CurvedScalarWave/Actions/NumericInitialData.hpp

```diff
@@ -0,0 +1,234 @@
+  NumericInitialData(const NumericInitialData& rhs) = default;
+  NumericInitialData& operator=(const NumericInitialData& rhs) = default;
+  NumericInitialData(NumericInitialData&& /*rhs*/) = default;
+  NumericInitialData& operator=(NumericInitialData&& /*rhs*/) = default;
+  ~NumericInitialData() override = default;
+
+  /// \cond
+  explicit NumericInitialData(CkMigrateMessage* msg);
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(NumericInitialData);
+  /// \endcond
+
+  std::unique_ptr<evolution::initial_data::InitialData> get_clone()
+      const override {
+    return std::make_unique<NumericInitialData>(*this);
+  }
+
+  NumericInitialData(
+      std::string file_glob, std::string subfile_name,
+      std::variant<double, importers::ObservationSelector> observation_value,
+      std::optional<double> observation_value_epsilon,
+      bool enable_interpolation, ScalarVars selected_variables);
+
+  const importers::ImporterOptions& importer_options() const;
+
+  const ScalarVars& selected_variables() const { return selected_variables_; }
+
+  size_t volume_data_id() const;
+
+  template <typename... AllTags>
+  void select_for_import(
+      const gsl::not_null<tuples::TaggedTuple<AllTags...>*> fields) const {
+    // Select the subset of the available variables that we want to read from
+    // the volume data file
+    using selected_vars = std::decay_t<decltype(selected_variables_)>;
+    tmpl::for_each<typename selected_vars::tags_list>(
+        [&fields, this](const auto tag_v) {
+          using tag = typename std::decay_t<decltype(tag_v)>::type::tag;
+          get<importers::Tags::Selected<tag>>(*fields) =
+              get<VarName<tag>>(selected_variables_);
+        });
+  }
+
+  template <typename... AllTags>
+  void set_initial_data(const gsl::not_null<Scalar<DataVector>*> psi_scalar,
+                        const gsl::not_null<Scalar<DataVector>*> pi_scalar,
+                        const gsl::not_null<tnsr::i<DataVector, 3>*> phi_scalar,
+                        const gsl::not_null<tuples::TaggedTuple<AllTags...>*>
+                            numeric_data) const {
+    *psi_scalar = std::move(get<CurvedScalarWave::Tags::Psi>(*numeric_data));
+    *pi_scalar = std::move(get<CurvedScalarWave::Tags::Pi>(*numeric_data));
+    *phi_scalar = std::move(get<CurvedScalarWave::Tags::Phi<3>>(*numeric_data));
+  }
+
+  void pup(PUP::er& p) override;
+
+  friend bool operator==(const NumericInitialData& lhs,
+                         const NumericInitialData& rhs);
+
+ private:
+  importers::ImporterOptions importer_options_{};
+  ScalarVars selected_variables_{};
+};
+
+namespace Actions {
+
+/*!
+ * \brief Dispatch loading numeric initial data from files.
+ *
+ * Place this action before
+ * CurvedScalarWave::Actions::SetNumericInitialData in the action list.
+ * See importers::Actions::ReadAllVolumeDataAndDistribute for details, which is
+ * invoked by this action.
+ */
+struct ReadNumericInitialData {
```

Comment:
Have you actually enabled this in an executable? Note that the GH and GHMHD executables use an action called `SetInitialData` that looks at the option-created initial data class (numeric or analytic) and based on that decides whether to load numeric ID or evaluate analytic ID. See `src/Evolution/Systems/GeneralizedHarmonic/Actions/SetInitialData.hpp`.